app/controllers/review_item/reject.js

```diff
@@ -46,120 +46,159 @@ export default Ember.Controller.extend({
   selectedId: Ember.computed("rejectionReasonId", {
     get: function() {
       this.set("isBlank", false);
-      var reasonId = this.get('rejectionReasonId');
-      if(reasonId) { return reasonId; }
-      else {
-        if(this.get("rejectReason") && this.get("rejectReason").length > 0) {
+      var reasonId = this.get("rejectionReasonId");
+      if (reasonId) {
+        return reasonId;
+      } else {
+        if (this.get("rejectReason") && this.get("rejectReason").length) {
           return "-1";
         }
       }
     },
     set: function(key, value) {
       this.set("isBlank", false);
-      this.set('noReasonSelected', false);
+      this.set("noReasonSelected", false);
       return value;
     }
   }),
 
-  rejectionOptions: Ember.computed(function(){
-    return this.store.peekAll('rejection_reason').sortBy('id');
+  rejectionOptions: Ember.computed(function() {
+    return this.store.peekAll("rejection_reason").sortBy("id");
   }),
 
-  cannotSave(){
-    var pkgs = this.store.peekRecord("item", this.get("itemId")).get("packages");
-    if(pkgs && pkgs.length > 0 && (pkgs.get("firstObject.hasAllPackagesDesignated") || pkgs.get("firstObject.hasAllPackagesDispatched"))){
-      this.get('messageBox').alert(this.get("i18n").t('designated_dispatched_error'));
+  cannotSave() {
+    var pkgs = this.store
+      .peekRecord("item", this.get("itemId"))
+      .get("packages");
+    if (
+      pkgs &&
+      pkgs.length &&
+      (pkgs.get("firstObject.hasAllPackagesDesignated") ||
+        pkgs.get("firstObject.hasAllPackagesDispatched"))
+    ) {
+      this.get("messageBox").alert(
+        this.get("i18n").t("designated_dispatched_error")
+      );
       return true;
     }
     return false;
   },
 
-  rejectValidation(selectedReason,rejectProperties){
-    if(selectedReason === undefined) {
-      this.set('noReasonSelected', true);
+  rejectValidation(selectedReason, rejectProperties) {
+    if (selectedReason === undefined) {
+      this.set("noReasonSelected", true);
       return false;
     }
-    if(selectedReason === "-1" && Ember.$.trim(rejectProperties.rejectReason).length === 0) {
+    if (
+      selectedReason === "-1" &&
+      Ember.$.trim(rejectProperties.rejectReason).length === 0
+    ) {
       this.set("isBlank", true);
       return false;
     }
     return true;
   },
 
-  actions: {
+  rejectProperties() {
+    let rejectProperties = this.getProperties("rejectReason");
```

Comment:
I think it would be better if we could write something like this
```suggestion
    let rejectProperties = this.getProperties("rejectReason");
    return {
      rejectResons: rejectProperties,
      state_event: "reject",
      //similarly other properties, it becomes more understandable what this object has 
    }
```
This is my view :) let me know your thought.

Reply:
Sure @abulsayyad123 :)